web_ui/src/shared/components/media-items-list/media-items-list.component.tsx

```diff
@@ -1,52 +1,84 @@
+    endReached?: () => void;
+    itemContent: (item: T) => ReactNode;
+    idFormatter: (item: T) => string;
 }
 
-// Each thumbnail has approximately 150 pixel height (this differs per resolution
-// as well as the large/medium/small thumbnails).
-// We want to overscan a few (4) rows of thumbnails so that when the user scrolls
-// they won't have to wait for loading the images of the next rows
-const OVERSCAN = 150 * 4;
-
-export const MediaItemsList = forwardRef<VirtuosoGridHandle | null, MediaItemsListProps>(
-    ({ id, viewMode, totalCount, itemContent, endReached, ariaLabel, ...styleProps }, ref): JSX.Element => {
-        const style = useMemo(() => {
-            return { ...VIEW_MODE_SETTINGS[viewMode] };
-        }, [viewMode]);
-
-        return (
-            <View id={id} height={'100%'} width={'100%'} {...styleProps}>
-                <VirtuosoGrid
-                    aria-label={ariaLabel}
+export const MediaItemsList = <T,>({
+    id,
+    height,
+    viewMode,
+    ariaLabel,
+    mediaItems,
+    viewModeSettings = VIEW_MODE_SETTINGS,
+    itemContent,
+    endReached,
+    idFormatter,
+}: MediaItemsListProps<T>): JSX.Element => {
+    const config = viewModeSettings[viewMode];
+    const isDetails = viewMode === ViewModes.DETAILS;
+    const formattedMediaItems = mediaItems?.map((item) => ({ id: idFormatter(item), ...item })) ?? [];
+
+    const ref = useRef(null);
+    useLoadMore({ onLoadMore: endReached }, ref);
+
+    const layoutOptions = isDetails
+        ? {
+              gap: config.gap,
+              rowHeight: config.size,
+          }
+        : {
+              minSpace: new Size(config.gap, config.gap),
+              minItemSize: new Size(config.minItemSize, config.minItemSize),
+              maxColumns: config.maxColumns,
+              preserveAspectRatio: true,
+          };
+
+    return (
+        <View id={id} UNSAFE_className={classes.mainContainer} height={height}>
+            <Virtualizer layout={isDetails ? ListLayout : GridLayout} layoutOptions={layoutOptions}>
+                <AriaComponentsListBox
                     ref={ref}
-                    totalCount={totalCount}
-                    overscan={OVERSCAN}
-                    style={style}
-                    className={classes.gridListContainer}
-                    listClassName={viewMode === ViewModes.DETAILS ? undefined : classes.gridList}
-                    itemClassName={viewMode === ViewModes.DETAILS ? undefined : classes.gridItem}
-                    itemContent={itemContent}
-                    endReached={endReached}
-                />
-            </View>
-        );
-    }
-);
+                    layout={isDetails ? 'stack' : 'grid'}
+                    aria-label={ariaLabel}
+                    items={formattedMediaItems}
+                    className={classes.container}
+                >
+                    {(item) => (
+                        <ListBoxItem textValue={item.id} style={{ height: '100%' }}>
```

Comment:
Can we use a non-id for the text value? E.g. perhaps we can pass in the name of the item?

Reply:
Since the data isn't consistent with the `name` property, I added a `getTextValue` handler to allow for a more flexible implementation.